gama/genetic_programming/operations.py

```diff
@@ -1,44 +1,323 @@
+                temp_config_space.get_hyperparameter(
+                    temp_config_space.meta[output_type]
+                ),
+                exclude.__str__(),
+            )
+        )
+        config = temp_config_space.sample_configuration()
+    else:
+        config = config_space.sample_configuration()
+
+    if output_type in [DATA_TERMINAL, "preprocessors"]:
+        (
+            preprocessor_primitive,
+            preprocessor_terminals,
+        ) = _config_preprocessor_to_primitive_node(
+            config, config_space.meta, config_space.get_conditions()
+        )
+        return PrimitiveNode(
+            preprocessor_primitive,
+            data_node=DATA_TERMINAL,
+            terminals=preprocessor_terminals,
+        )
+    estimator_primitive, estimator_terminals = _config_estimator_to_primitive_node(
+        config, config_space.meta, config_space.get_conditions()
+    )
+    return PrimitiveNode(
+        primitive=estimator_primitive,
+        data_node=DATA_TERMINAL,
+        terminals=estimator_terminals,
+    )
 
 
 def create_random_expression(
-    primitive_set: dict, min_length: int = 1, max_length: int = 3
+    config_space: cs.ConfigurationSpace,
+    min_length: int = 1,
+    max_length: int = 3,
 ) -> PrimitiveNode:
     """Create at least min_length and at most max_length chained PrimitiveNodes."""
     individual_length = random.randint(min_length, max_length)
-    learner_node = random_primitive_node(
-        output_type="prediction", primitive_set=primitive_set
+    return _config_to_primitive_node(
+        config=config_space.sample_configuration(),
+        config_meta=config_space.meta,
+        conditions=config_space.get_conditions(),
+        config_length=individual_length,
+    )
+
+
+def extract_valid_hyperparameters(
+    cond: cs.conditions, config: cs.Configuration, config_meta: dict, meta_key: str
+) -> Union[str, None]:
+    """Extract valid hyperparameters from a condition.
+
+    For each supported ConfigSpace condition type, extract the valid hyperparameters
+    from the condition. The valid hyperparameters are the hyperparameters that are
+    valid for the given condition and configuration based on the meta_key.
+
+    Supported ConfigSpace condition types:
+    - EqualsCondition
+    - AndConjunction
+
+    Readers are encouraged to add support for more ConfigSpace condition types if
+    needed. Open an issue or pull request on the GAMA GitHub repository.
+
+    Parameters
+    ----------
+    cond : cs.conditions
+        A condition of type ConfigSpace.
+    config : cs.Configuration
+        A configuration of type ConfigSpace.
+    config_meta : dict
+        The meta information of the ConfigSpace.
+    meta_key : str
+        The meta key of the ConfigSpace.
+    """
+    if meta_key not in config_meta:
+        raise ValueError(f"Meta key {meta_key} not in config_meta")
+    if type(cond) not in [cs.conditions.EqualsCondition, cs.conditions.AndConjunction]:
```

Comment:
```suggestion
    if not isinstance(cond, (cs.conditions.EqualsCondition, cs.conditions.AndConjunction)):
```

Reply:
or even better: since you check them separately below, just use an `else` clause that raises this error